### PR TITLE
[SPARK-41121][BUILD] Upgrade `sbt-assembly` to 2.0.0

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This pr aims upgrade sbt-assembly plugin from 1.2.0 to 2.0.0

### Why are the changes needed?
Release notes as follows:
https://github.com/sbt/sbt-assembly/releases/tag/v2.0.0
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -25,7 +25,7 @@ libraryDependencies += "com.puppycrawl.tools" % "checkstyle" % "9.3"
 // checkstyle uses guava 31.0.1-jre.
 libraryDependencies += "com.google.guava" % "guava" % "31.0.1-jre"
 
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "1.2.0")
+addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "2.0.0")
```
Comment (LuciferYang) [inline project/plugins.sbt:28]: Can you help to check whether the contents of the `connect` assembly jar are consistent before and after the pr?

**Review by dongjoon-hyun — COMMENTED** [review]
Hi, @panbingkun . Is there any update according to @LuciferYang 's comment?

File: project/plugins.sbt (hunk 1 above)
Comment (panbingkun) [reply project/plugins.sbt:28]: They are consistent

> When 1.2.0:
image

> After 2.0.0:
image

**Comment by panbingkun** [comment]
> Hi, @panbingkun . Is there any update according to @LuciferYang 's comment?

They are consistent.
The comparison process is as above.
@LuciferYang @dongjoon-hyun

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Pull request closed without merging**